server/service/software.go

```diff
@@ -50,17 +50,27 @@ func listSoftwareEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 }
 
 type listSoftwareVersionsResponse struct {
-	Count           int              `json:"count"`
-	CountsUpdatedAt *time.Time       `json:"counts_updated_at"`
-	Software        []fleet.Software `json:"software,omitempty"`
-	Err             error            `json:"error,omitempty"`
+	Count           int                       `json:"count"`
+	CountsUpdatedAt *time.Time                `json:"counts_updated_at"`
+	Software        []fleet.Software          `json:"software,omitempty"`
+	Meta            *fleet.PaginationMetadata `json:"meta"`
+	Err             error                     `json:"error,omitempty"`
 }
 
 func (r listSoftwareVersionsResponse) error() error { return r.Err }
 
 func listSoftwareVersionsEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
 	req := request.(*listSoftwareRequest)
-	resp, err := svc.ListSoftware(ctx, req.SoftwareListOptions)
+
+	// always include pagination for new software versions endpoint (not included by default in
+	// legacy endpoint for backwards compatibility)
+	opt := &req.SoftwareListOptions
+	opt.ListOptions.IncludeMetadata = true
+	if opt.ListOptions.PerPage == 0 {
+		opt.ListOptions.PerPage = 20
```

Comment:
Any reason we set an low default per page here? Typically, `fleetctl` doesn't pass any per page limit, and gets as many as it can from the server. This will drastically reduce the number of results that `fleetctl get software --versions` displays.